### PR TITLE
serviceDiscovery has been added.

### PR DESCRIPTION
**Pull request by maksimfedin**

### CODE REVIEW

**Comment by maksimfedin** [comment]
@apolubelov pls consider to merge.

File: src/main/scala/org/enterprisedlt/fabric/client/FabricChannel.scala [hunk 1]
```diff
@@ -12,16 +14,24 @@ class FabricChannel(
     fabricClient: HFClient,
     fabricChannel: Channel
 ) {
-    def getChainCode(name: String, codec: BinaryCodec /* TODO: , endorsementTimeout: Int = */): FabricChainCode =
+    def getChainCode(name: String, codec: BinaryCodec, chaincodeServiceDiscovery: Boolean /* TODO: , endorsementTimeout: Int = */): FabricChainCode =
```
Comment (apolubelov) [inline src/main/scala/org/enterprisedlt/fabric/client/FabricChannel.scala]: put default false for discovery

File: src/main/scala/org/enterprisedlt/fabric/client/FabricChannel.scala [hunk 2]
```diff
@@ -12,16 +14,24 @@ class FabricChannel(
...
+            codec,
+            chaincodeServiceDiscovery
         )
 
     //=========================================================================
     def setupBlockListener(listener: BlockListener): Try[String] = Try {
         fabricChannel.registerBlockListener(listener)
     }
+
+    def addClientCerts(clientCertFile: String, clientKeyFile: String): Unit = {
```
Comment (apolubelov) [inline src/main/scala/org/enterprisedlt/fabric/client/FabricChannel.scala]: should be in constructor

File: src/main/scala/org/enterprisedlt/fabric/client/FabricClient.scala [hunk 3]
```diff
@@ -31,7 +34,17 @@ class FabricClient(
...
+                channel.addPeer(mkPeer(config))
+            } else {
+                val peerOptions = createPeerOptions
+                  .setPeerRoles(util.EnumSet.of(
+                      PeerRole.SERVICE_DISCOVERY,
+                      PeerRole.LEDGER_QUERY,
+                      PeerRole.EVENT_SOURCE,
+                      PeerRole.CHAINCODE_QUERY))
+                channel.addPeer(mkPeer(config), peerOptions)
+            }
```
Comment (apolubelov) [inline src/main/scala/org/enterprisedlt/fabric/client/FabricClient.scala]: the role has to be in peer config

File: src/main/scala/org/enterprisedlt/fabric/client/configuration/PeerConfig.scala [hunk 4]
```diff
@@ -6,5 +6,6 @@ package org.enterprisedlt.fabric.client.configuration
 case class PeerConfig(
     name: String,
     address: String,
+    peerServiceDiscovery: Boolean,
```
Comment (apolubelov) [inline src/main/scala/org/enterprisedlt/fabric/client/configuration/PeerConfig.scala]: model roles here instead of just discovery flag

**Review by apolubelov — CHANGES_REQUESTED** [review]

File: src/main/scala/org/enterprisedlt/fabric/client/FabricChannel.scala (hunk 2 above)
Comment (maksimfedin) [reply src/main/scala/org/enterprisedlt/fabric/client/FabricChannel.scala]: I'm going to remove this method.

File: src/main/scala/org/enterprisedlt/fabric/client/FabricClient.scala (hunk 3 above)
Comment (maksimfedin) [reply src/main/scala/org/enterprisedlt/fabric/client/FabricClient.scala]: done.

File: src/main/scala/org/enterprisedlt/fabric/client/configuration/PeerConfig.scala (hunk 4 above)
Comment (maksimfedin) [reply src/main/scala/org/enterprisedlt/fabric/client/configuration/PeerConfig.scala]: done.

File: src/main/scala/org/enterprisedlt/fabric/client/FabricChannel.scala (hunk 1 above)
Comment (maksimfedin) [reply src/main/scala/org/enterprisedlt/fabric/client/FabricChannel.scala]: done.

**Review by apolubelov — APPROVED** [review]

**Pull request merged**